Document that *become* vars not available for templating (#72870)
ansible/ansible: docs/docsite/rst/user_guide/become.rst

to running specific command paths only, use Ansible from a special account that
does not have this constraint, or use :ref:`ansible_tower` to manage indirect access to SSH credentials.

> Become variables not available for templating

Reviewer: ```suggestion
Become variables and templating
```

---------------------------------------------

> As a security precaution, Ansible ``*become*``  variables such as

Reviewer 2: This is still incorrect, also it assumes tha tthe CLI/keywords or configuration set ansible_become variables, they DO NOT and never did.

We sometimes populate those variables based on the 'resolved value' but are trying to avoid it going forward for security reasons . This does not mean the variables are removed from being templatable, it only means we avoid defining them in certain cases, if the variable is defined, it is available for templating.  

The variable itself is ONE way of defining the value to be used, at some points it was hijacked to also 'reflect' the resolved value but this creates 2 problems, one is security, the other is 'rentrant values' which would bring the incorrect 'resolved value' as the variable itself has high precedence and changes meant to change the value would be incorrectly overridden. 

For example, you use the `--become-user` option to set it to `foo`, if we set the `ansible_become_user` variable to `foo` then a play that then uses the keyword `become_user` to override the value to `bar` would not work (since variable has greater precedence then keyword) which would not be what the play author expected (since keyword has greater precedence than CLI)

> As a security precaution, Ansible ``*become*`` magic variables such as

Reviewer: ```suggestion
As a security precaution, Ansible ``*become*``  variables such as
```

> ``ansible_become_pass`` are *not* available for templating. For instance, even

Reviewer: ```suggestion
``ansible_become_pass`` are *not* available for templating if set with the CLI, keywords, or configuration. For instance, 
```

if using ``--ask-become-pass``/``-K`` or ``--become-user`` from the command
line, ``{{ ansible_become_pass }}`` and ``{{ ansible_become_method }}`` will
not be available to use in template strings.

> 

Reviewer: ```suggestion
You can access these variables in template strings if you set the variable in other ways.

.. code-block:: text

    ansible -m debug -a 'msg={{ansible_become_password}}' -e ansible_become_password=test2 localhost

```
